docs/data/charts/bars/BarScatterCompostion.js

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import { BarPlot } from '@mui/x-charts/BarChart';
+import { ScatterPlot } from '@mui/x-charts/ScatterChart';
+import { ChartsXAxis } from '@mui/x-charts/ChartsXAxis';
+import { ChartsYAxis } from '@mui/x-charts/ChartsYAxis';
+import { ChartsLegend } from '@mui/x-charts/ChartsLegend';
+import { ChartsGrid } from '@mui/x-charts/ChartsGrid';
+import { ChartDataProvider } from '@mui/x-charts/ChartDataProvider';
+import { ChartsSurface } from '@mui/x-charts/ChartsSurface';
+import { ChartsTooltip } from '@mui/x-charts/ChartsTooltip';
+import { GDPdata } from '../dataset/gdpGrowth';
+
+const chartSetting = {
+  xAxis: [
+    {
+      label: 'Value',
+      colorMap: {
+        type: 'piecewise',
+        thresholds: [0],
+        colors: ['#ff4d4f', '#1976d2'],
+      },
+    },
+  ],
```

Comment:
You can create a second xAxis to use for the scatter only, and then set a custom color for it in the scatter series itself

Reply:
This worked thanks, but now there is a new problem. I added piecewise legend (in JS code) which removed the other legend from the Scatter. The TS code shows legends from both series but without piecewise demarcation.

Reply:
You will need a custom legend to have both piecewise and series.

Reply:
Quite cool! I'll use the second one.

Reply:
I couldn't figure out in which component should I pass this CSS selector. I tried passing it in ChartsDataProvider, but it doesn't work. Need your input.
```
sx = {{ [`[data-series="bar"] .${legendClasses.mark} rect`]: {
          fill: 'url(#diagonalGradient)',
        },
}}
```

Reply:
You can put it on the Box